### PR TITLE
NERT-644 Application report in place, and some fixes to the dashboard

### PR DESCRIPTION
**Pull request by oscar-ip**
## Links to Jira Tickets

- NERT-644 NERT-645 NERT-646 NERT-647

## Description of Changes

- Application report is now working
- New API calls in place
- Additional changes and fixes to dashboard when there is no project nor plans

Note: currently all reports are routed to the Application report

### CODE REVIEW

File: api/src/paths/reports/appreport/view.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { SYSTEM_ROLE } from '../../../constants/roles';
+import { getDBConnection } from '../../../database/db';
+import { authorizeRequestHandler } from '../../../request-handlers/security/authorization';
+import { ReportService } from '../../../services/reports-service';
+import { getLogger } from '../../../utils/logger';
+
+const defaultLog = getLogger('paths/reports/appreport/view');
```
Comment (KjartanE) [inline api/src/paths/reports/appreport/view.ts:9]: could be renamed to something a bit more descriptive. ie: `paths/reports/app/view` maybe? appreport seems confusing

File: api/src/services/reports-service.ts [hunk 2]
```diff
@@ -22,4 +22,16 @@ export class ReportService extends DBService {
...
+  /**
+   * Get application report data.
+   *
+   * @return {*}  {Promise<IGetAppReport>}
+   * @memberof ReportService
+   */
+  async getAppReport(): Promise<IGetAppReport[]> {
+    const reportAppResponse = await Promise.all([this.reportRepository.getAppReportData()]);
+
+    return reportAppResponse[0];
```
Comment (KjartanE) [inline api/src/services/reports-service.ts:35]: seems random to select the first here. Would be better suited to select the first in the SQL or just select the `row[0]` on return or error out if not found

File: app/src/components/dialog/InfoDialog.tsx [hunk 3]
```diff
@@ -55,7 +55,7 @@ const InfoDialog: React.FC<IInfoDialogProps> = (props) => {
   };
 
   const handleError = (error: any) => {
-    console.error('Error playing media', error);
+    console.error('Unable to play media', error);
```
Comment (KjartanE) [inline app/src/components/dialog/InfoDialog.tsx:58]: should probably remove console log

File: api/src/paths/reports/appreport/view.ts (hunk 1 above)
Comment (oscar-ip) [reply api/src/paths/reports/appreport/view.ts:9]: it will change, for now I left like that

File: api/src/services/reports-service.ts (hunk 2 above)
Comment (oscar-ip) [reply api/src/services/reports-service.ts:35]: See the response from the query, it is a single query that returns an array of objects 
initially I had it as row[0] but it was not ion the response.

File: app/src/components/dialog/InfoDialog.tsx (hunk 3 above)
Comment (oscar-ip) [reply app/src/components/dialog/InfoDialog.tsx:58]: It is there so we know which paths in the S3 are missing, probably we will remove once we figure out our interactions with the bcGov docs team as they will be providing content

**Review by KjartanE — APPROVED** [review]
🐱

**Pull request merged**